src/server/location/location.controller.ts

```diff
@@ -51,4 +52,15 @@ export class LocationController {
     : Promise<RoomMeetingResponse[]> {
     return this.locationService.getRooms(roomInfo);
   }
+
+  @Get('/admin')
+  @ApiOperation({ summary: 'Retrieve all room names and corresponding building and campus data' })
+  @ApiOkResponse({
+    type: RoomAdminResponse,
+    description: 'An array of all room names and corresponding building and campus data',
+    isArray: true,
+  })
+  public async getAdminRooms(): Promise<RoomAdminResponse[]> {
```

Comment:
As-per my other comment regarding method naming - unless there's a specific reason to do otherwise, I think `getRooms` or something more generic might be more appropriate....